streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -103,7 +105,8 @@ InternalTopologyBuilder builder() {
         return builder;
     }
 
-    void handleRebalanceStart(final Set<String> subscribedTopics) {
+    // visible for testing
+    public void handleRebalanceStart(final Set<String> subscribedTopics) {
```

Comment:
Hm...imo that seems like overkill for just this one method, but we'll likely need to expose more and more methods as KIP-441 goes on. WDYT about leaving this as-is for now, and adding an accessor class if we find ourselves making more methods public in later PRs?

Reply:
To be honest, this also bugs me :)

What if we move `org.apache.kafka.streams.tests.StreamsUpgradeTest.FutureStreamsPartitionAssignor` into package `org.apache.kafka.streams.processor.internals`? Then package-private would continue to work fine.

My personal bias is that anytime you see `// visible for testing`, you're looking at a potential bug, because _nothing_ prevents that comment from becoming false, and in fact, I have found such comments in our code base that were already false. Either this method is part of the public contract of the class, or it's not.

That said, if you really prefer it this way, we can keep it (although, I might ask you to review a clean-up PR later ;) )

Reply:
Alright alright, I concede. I have pretty strong (negative) feelings about the future partition assignor, which I'm sure I've expressed (ranted about) before, and don't see much value in moving it to a new package so it can use methods it really shouldn't need to call directly in the first place...
But that's a discussion for another time.  I'll add an accessor class and hold off on sprinkling soon-to-be-obselete comments throughout the code for now

Reply:
Ok actually I'm just going to fix the `FutureStreamsPartitionAssignor` to not try and call this method. Thanks ^^